ssg/boolean_expression.py

```diff
@@ -87,16 +99,28 @@ def as_id(self):
         id_str = self.name
         for (op, ver) in self.spec.specs:
             id_str += '_{0}_{1}'.format(SPEC_OP_ID_TRANSLATION.get(op, 'unknown_spec_op'), ver)
+        if self.spec.extras:
+            id_str += '_' + self.spec.extras[0]
         return id_str
 
+    def as_dict(self):
+        res = {'id': self.as_id(), 'name': self.name, 'arg': ''}
+        if self.spec.extras:
+            res['arg'] = self.spec.extras[0]
+        return res
+
     @property
-    def name(self):
-        return self.spec.project_name
+    def arg(self):
+        return self.spec.extras[0] if self.spec.extras else None
```

Comment:
This is completely unreadable - the method name gives no hint, and its body either.